Automattic/jetpack
# SEO Tools / Open Graph: improve cleaning up of description meta tags

Fixes #16147

#### Changes proposed in this Pull Request:

This PR introduces changes to 2 features:
- Meta descriptions for single posts when using Jetpack's SEO Tools feature.
- Open Graph Meta descriptions for single posts when using Jetpack's Sharing or Publicize feature.

#### Jetpack product discussion

* No product change

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Start from a brand new site with a Premium Plan
* Go to Jetpack > Settings > Sharing, and enable sharing buttons.
* Go to Jetpack > Settings > Traffic, and follow the link to Calypso to enable the SEO Tools feature
* Head over to Site > Posts in Calypso
* Create a new post there. In the beginning of that post, in the HTML view, add `<script>alert("hello");</script>` before some content. Publish your post.
* Visit your post, view source, and look for both the `<meta property="og:description"` tag and the `<meta name="description"` tag. 
* None of those tags should include the `alert` tag.
* Back to the post editor, try adding other elements to the post content and to the post excerpt, and make sure that the following is always true for the Open Graph Meta tags:

* There should be:
 * - no links
 * - no shortcodes
 * - no html tags or their contents
 * - not too many words.

#### Proposed changelog entry for your changes:

* SEO Tools / Open Graph: improve cleaning up of description meta tags
